modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/FeatureVectorFieldMapper.java

```diff
@@ -0,0 +1,182 @@
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        public Query existsQuery(QueryShardContext context) {
+            throw new UnsupportedOperationException("[feature_vector] fields do not support [exists] queries");
+        }
+
+        @Override
+        public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName) {
+            throw new UnsupportedOperationException("[feature_vector] fields do not support sorting, scripting or aggregating");
+        }
+
+        @Override
+        public Query termQuery(Object value, QueryShardContext context) {
+            throw new UnsupportedOperationException("Queries on [feature_vector] fields are not supported");
+        }
+    }
+
+    private FeatureVectorFieldMapper(String simpleName, MappedFieldType fieldType, MappedFieldType defaultFieldType,
+                                Settings indexSettings, MultiFields multiFields, CopyTo copyTo) {
+        super(simpleName, fieldType, defaultFieldType, indexSettings, multiFields, copyTo);
+        assert fieldType.indexOptions().compareTo(IndexOptions.DOCS_AND_FREQS) <= 0;
+    }
+
+    @Override
+    protected FeatureVectorFieldMapper clone() {
+        return (FeatureVectorFieldMapper) super.clone();
+    }
+
+    @Override
+    public FeatureVectorFieldType fieldType() {
+        return (FeatureVectorFieldType) super.fieldType();
+    }
+
+    @Override
+    public FieldMapper parse(ParseContext context) throws IOException {
+        if (context.externalValueSet()) {
+            throw new IllegalArgumentException("[feature_vector] fields can't be used in multi-fields");
+        }
+
+        if (context.parser().currentToken() != Token.START_OBJECT) {
+            throw new IllegalArgumentException("[feature_vector] fields must be json objects, expected a START_OBJECT but got: " +
+                    context.parser().currentToken());
+        }
+
+        String feature = null;
+        for (Token token = context.parser().nextToken(); token != Token.END_OBJECT; token = context.parser().nextToken()) {
+            if (token == Token.FIELD_NAME) {
+                feature = context.parser().currentName();
+            } else if (token == Token.VALUE_NULL) {
+                // ignore feature, this is consistent with numeric fields
+            } else if (token == Token.VALUE_NUMBER || token == Token.VALUE_STRING) {
+                final String key = name() + "." + feature;
+                float value = context.parser().floatValue(true);
+                if (context.doc().getByKey(key) != null) {
+                    throw new IllegalArgumentException("[feature_vector] fields do not support indexing multiple values for the same " +
+                            "feature [" + key + "] in the same document");
+                }
+                context.doc().addWithKey(key, new FeatureField(name(), feature, value));
```

Comment:
Did I understand correctly - if we index in ES:
```
"topics": {
	"politics": 20,
	"economics": 50
}
```
in Lucene, it will translate to a field with a name `topics`, with values:  "politics" with freq 20, and "economics" with freq 50?

Reply:
The actual term frequency will actually be `Float.floatToIntBits(20) >>> 15` rather than 20, but other than that this is correct. You can look at `FeatureField.tokenStream` for more details. So basically it is a `float` but with only 8 bits for the mantissa rather than 23.